filebeat/tests/system/test_crawler.py

```diff
@@ -675,3 +675,130 @@ def test_encodings(self):
         for _, _, text in encodings:
             assert text in lines
             assert text + " 2" in lines
+
+    def test_include_lines(self):
+        """
+        Checks if only the log lines defined by include_lines are exported
+        """
+
+        self.render_config_template(
+            path=os.path.abspath(self.working_dir) + "/log/*",
+            include_lines=["^ERR", "^WARN"]
+        )
+        os.mkdir(self.working_dir + "/log/")
+
+        testfile = self.working_dir + "/log/test.log"
+        file = open(testfile, 'w')
+
+        iterations = 20
+        for n in range(0, iterations):
+            file.write("DBG: a simple debug message" + str(n))
+            file.write("\n")
+            file.write("ERR: a simple error message" + str(n))
+            file.write("\n")
+            file.write("WARNING: a simple warning message" + str(n))
+            file.write("\n")
+
+        file.close()
+
+        filebeat = self.start_filebeat()
+
+        self.wait_until(
+            lambda: self.log_contains(
```

Comment:
More recently we started to prefer using conditions like `self.output_has(lines=40)`. Seems to result in less flaky tests.